utils/nctl/sh/staging/stage.sh

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env bash
+
+source "$NCTL/sh/utils/main.sh"
+
+# ----------------------------------------------------------------
+# MAIN
+# ----------------------------------------------------------------
+
+#######################################
+# Initialises scenario binaries.
+# Arguments:
+#   Scenario ordinal identifier.
+#######################################
+function _set_assets()
+{
+    local PATH_TO_STAGING=${1}
+    local PATH_TO_STAGING_VERSION
+    local SCENARIO_COMMIT_HASH
+    local SCENARIO_VERSION
+    local STAGING_TARGET
+    local UPGRADE_STAGING_TARGET
+    local IFS
+
+    log "... setting assets"
+
+    pushd "$PATH_TO_STAGING/src" || exit
+
+    for UPGRADE_STAGING_TARGET in "${NCTL_STAGING_TARGETS[@]}"
+    do        
+        # Destructure settings.
+        IFS=':'
+        read -ra STAGING_TARGET <<< "$UPGRADE_STAGING_TARGET"
+        SCENARIO_COMMIT_HASH="${STAGING_TARGET[1]}"
+        SCENARIO_VERSION="${STAGING_TARGET[0]}"
+        PATH_TO_STAGING_VERSION="$PATH_TO_STAGING/$SCENARIO_VERSION"
+        log "... ... $SCENARIO_VERSION"
+
+        # Set commit.
+        git checkout "$SCENARIO_COMMIT_HASH" > /dev/null 2>&1
+
+        # Compile binaries.
+        # if [ "$NCTL_COMPILE_TARGET" = "debug" ]; then
+        #     cargo build --package casper-node
+        #     cargo build --package casper-client
+        # else
+        #     cargo build --release --package casper-node
+        #     cargo build --release --package casper-client
+        # fi
+        # make build-contract-rs/add-bid
+        # make build-contract-rs/delegate
+        # make build-contract-rs/transfer-to-account-u512
+        # make build-contract-rs/undelegate
+        # make build-contract-rs/withdraw-bid
+        # make build-contract-rs/activate-bid
+
+        # Copy binaries -> staging.
+        if [ "$NCTL_COMPILE_TARGET" = "debug" ]; then
+            cp "./target/debug/casper-node" "$PATH_TO_STAGING_VERSION/bin"
```

Comment:
These paths fail for me.  They need something like `$NCTL_CASPER_HOME` prepended, or if you use the suggestion about per-commit build dirs, then that path will need to be provided here too.

Same for all paths here starting `./target`, and the `./resources` paths just need `$NCTL_CASPER_HOME` prepended.